freesound/middleware.py

```diff
@@ -77,31 +63,45 @@ def __call__(self, request):
                 cache.set(cache_key, [has_old_license, has_sounds], 2592000) # 30 days cache
                 if has_old_license and has_sounds:
                     return HttpResponseRedirect(reverse("bulk-license-change"))
-
-            else :
+            else:
                 has_old_license = cache_info[0]
                 has_sounds = cache_info[1]
-                #print "CACHE LICENSE: has_old_license=" + str(has_old_license) + " has_sounds=" + str(has_sounds)
                 if has_old_license and has_sounds:
                     return HttpResponseRedirect(reverse("bulk-license-change"))
         response = self.get_response(request)
         return response
 
 
+class FrontendPreferenceHandler(object):
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        """
+        This middleware sets a session variable when the parameter 'new_frontend' is received.
+        The 'render' method will use this session variable to display the new/old frontend
+        """
+        if request.GET.get(settings.FRONTEND_CHOOSER_REQ_PARAM_NAME, None):
+            request.session[settings.FRONTEND_SESSION_PARAM_NAME] = \
+                request.GET.get(settings.FRONTEND_CHOOSER_REQ_PARAM_NAME)
+        response = self.get_response(request)
+        return response
+
+
 class TosAcceptanceHandler(object):
     def __init__(self, get_response):
         self.get_response = get_response
 
     def __call__(self, request):
 
         if request.user.is_authenticated \
-            and not 'tosacceptance' in request.get_full_path() \
-            and not 'logout' in request.get_full_path() \
-            and not 'tos_api' in request.get_full_path() \
-            and not 'tos_web' in request.get_full_path() \
-            and not 'contact' in request.get_full_path() \
-            and not 'bulklicensechange' in request.get_full_path() \
-            and not request.get_full_path().startswith(settings.MEDIA_URL):
+                and 'tosacceptance' not in request.get_full_path() \
```

Comment:
I don't like mixing features and refactoring in the same PR, but whatever :)

Reply:
I know, but we also have to be practical and improve things.